tests/python_package_test/test_dask.py

```diff
@@ -234,26 +234,34 @@ def test_classifier_pred_contrib(output, centers, client, listen_port):
     client.close(timeout=CLIENT_CLOSE_TIMEOUT)
 
 
+def test_different_ports_on_local_cluster(client):
+    for _ in range(5):
+        worker_address_to_port = client.run(lgb.dask._find_random_open_port)
+        found_ports = worker_address_to_port.values()
+        assert len(set(found_ports)) == len(found_ports)
+
+
 def test_training_does_not_fail_on_port_conflicts(client):
     _, _, _, dX, dy, dw = _create_data('classification', output='array')
 
     with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
         s.bind(('127.0.0.1', 12400))
 
-        dask_classifier = lgb.DaskLGBMClassifier(
-            time_out=5,
-            local_listen_port=12400,
-            n_estimators=5,
-            num_leaves=5
-        )
-        for _ in range(5):
-            dask_classifier.fit(
-                X=dX,
-                y=dy,
-                sample_weight=dw,
-                client=client
+        for local_listen_port in (None, 12400, 13000):
```

Comment:
could you please push this up into a `pytest.mark.parametrize`? See https://github.com/microsoft/LightGBM/blob/56fc036def837aa6a754ceb5cdecde212892edcd/tests/python_package_test/test_dask.py#L341 for an example if you've never used that before.